CITE Participation Guide

weirdscience opened the pull request:
This PR fixes #385 and #389 to host the CITE Participation Guide on the FPKI Guides website. This document was revised based on feedback from a Technical Working Group meeting and a three week community comment period.

Commit:
CITE Guide Upload

CITE participation guide converted to md.

Commit:
Add header

Add header and update OID tables

Commit:
Update Permalink

Update permalink

Commit:
Merge pull request #372 from GSA/staging

Merge to latest

Commit:
Merge pull request #386 from GSA/staging

Update

Commit:
Review

Update section 1 and 2

Commit:
Review 2

Update

Commit:
Fix OID tables

Commit:
Update OID Table

Commit:
Fix OID Table

Commit:
Fix  website table and remove testing schedule

Commit:
Fix header

Commit:
Fix links

Commit:
Final Draft

Reviewed and approved by TWG Chair.

Commit:
Fix bulleted list

Commit:
Editorial

Commit:
Test no table

Commit:
Fix Table

Commit:
Update CA DN Requirements

Update based on Treasury comment to allow "Development" in lieu of "Test" in the CA name.

Commit:
Fix Permalink

Commit:
Fix Header Formats

Commit:
Incorporate Treasury Comments

Fixes #389

Commit:
Merge pull request #402 from GSA/staging

Merge

Commit:
Fix Test Reqs and OID Table

Consolidate test requirements and insert placeholder for IdenTrust OID table

Commit:
Editorial and CleanUp

Commit:
Edits

Commit:
Fix spacing

Commit:
Fix Wells Fargo Table

lachellel commented:
@weirdscience I'm going to add some embedded comments to address; 

- Plain Language and active voice
- Formatting
- Alignment with the other pages (linking, headers)
- Content

I'll try to do a PR too to help address some of the above!

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+permalink: /tools/citeguide/
+---
+
+# Community Interoperability Test Environment (CITE) Participation Guidance
```
Remove this header - in the templates, the title is displayed first and having an additional top level header creates **two** headers for the page

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+
+This document provides guidelines and specifications for an FPKI Partner to participate in CITE. CITE Participants refer to an FPKI Partner establishing a test PKI certified or cross-certified with the Test Common Policy or Test Bridge CA. This document shall not define how CITE Participants conduct testing.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
```
Unnecessary - this is not a Standard or a policy doc.  It's a guide.  Align words with active voice.

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+Modifications to this document and any waivers are controlled by the FPKI Technical Working Group. <br>
```
What is a waiver in this context?  is there a compliance mechanism that requires a waiver to be granted for auditing purposes or other?

weirdscience replied:
Removed

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+
+When testing is successful in CITE, assurance is gained that the proposed change(s) will operate in the production FPKI as intended. When tests fail in the CITE, issues are identified and addressed without production FPKI impact. <br>
+
+## Terms and Conditions
```
This isn't a legal agreement - probably should  use other words than "Terms and Conditions"

weirdscience replied:
Done.

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+Last Update: February 12th, 2019 <br>
+
+## Overview
+
```
Need to have links to each section below for easy navigation.

weirdscience replied:
Added.

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+
+{% include alert-warning.html content="CITE Participant technical support is only available for scheduled testing with any outage resolved on a best effort basis. " %} <br>
+
+## Test Websites
```
If there is none - don't list at all.

weirdscience replied:
I like to keep it as a placeholder if an affiliate wants to share one, we know where to put it.

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+3. [Federal Shared Service Provider (SSP) Test OIDs](#federal-shared-service-provider-ssp-test-oids)
+4. [Non-Federal Issuer (NFI) Test OIDs](#non-federal-issuer-nfi-test-oids)
+5. [Commercial PKI Bridge Test OIDs](#commercial-pki-bridge-test-oids)
+6. [Deprecated Test OIDs](#deprecated-test-oids)
```
you had a question about deprecated test oids in your notices - if the deprecated test oids are not being used, remove from the list.  Too much information!

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+| 2.16.840.1.101.3.2.1.15.34 | id-dhs-certpcy-high | 2.16.840.1.101.3.2.1.15.4 |
+| 2.16.840.1.101.3.2.1.15.35 | id-dhs-certpcy-mediumHardware | 2.16.840.1.101.3.2.1.15.5 |
+
+#### Department of Justice (DOJ)
```
Justice OIDs should no longer be active?  No cross-certs in prod.

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+| 2.16.840.1.101.3.2.1.48.112 | treasury-pivi-cardAuth | 2.16.840.1.101.3.2.1.5.11 |
+| 2.16.840.1.101.3.2.1.48.113 | treasury-pivi-contentSigning | 2.16.840.1.101.3.2.1.5.12 |
+
+#### Federal Bureau of Investigation (FBI)
```
FBI no longer active?  No cross-certs in prod?

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+| --------------------------- | ---------------------- | ------------------ |
+| 2.16.840.1.101.3.2.1.48.37 | id-gpo-medium | 2.16.840.1.101.3.2.1.17.1 |
+
+#### Internal Revenue Service (IRS)
```
IRS no longer active?  no cross-certs in prod?

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+
+### Non-Federal Issuer (NFI) Test OIDs
+
+#### Entrust Management Services
```
Entrust **Managed** Services

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+| 2.16.840.1.114412.99.4.5.1 | Level 4 PIV-I - Hardware | 2.16.840.1.114412.4.5.1 |
+| 2.16.840.1.114412.99.4.5.2 | DigiCert Level 4 PIV-I - Card Authentication | 2.16.840.1.114412.4.5.2 |
+| 2.16.840.1.114412.99.4.5.3 | DigiCert Level 4 PIV-I - Content Signing | 2.16.840.1.114412.4.5.3 |
+| 2.16.840.1.114412.99.1.11 | DigiCert OV SSL | 2.16.840.1.114412.1.11 |
```
We shouldn't have an OV SSL mapped anymore (pending March). ?

weirdscience replied:
Done

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,459 @@
+| 1.3.6.1.4.1.24019.1.1.1.114 | CertiPath variant medium CBP Hardware | 1.3.6.1.4.1.24019.1.1.1.21 |
+| 1.3.6.1.4.1.24019.1.1.1.115 | CertiPath variant high CBP Hardware | 1.3.6.1.4.1.24019.1.1.1.22 |
+
+### Deprecated Test OIDS
```
I'd just remove all these deprecated test oids.  You can track somewhere for historical purposes but they are irrelevant to current?

weirdscience commented:
Thank you @lachellel . I'll get started on the comments.

Commit:
OID Table Update

Remove deprecated and non-prod OID tables.

Commit:
Integrate Comments

- Removed Title.
- Added links to each section for easy navigation.
- Integrated portions of document scope into overview and technical specifications and removed the section.
- Removed waiver requirement as this is a practice guide.
- Integrated portions of terms and conditions into technical specifications and removed the section.

Commit:
Fix section headers

Commit:
Correct Appendix Header

Commit:
Header fix

Commit:
Fix Appendix Header

Commit:
Update for IdenTrust OIDs

Commit:
Fix IdenTrust Table

lachellel commented:
@weirdscience 

do you want a review and a merge?  (not sure if this PR is ready or still waiting for edits!)

Commit:
Merge pull request #444 from GSA/staging

Update branch

Commit:
Fix IdenTrust Table

weirdscience commented:
@lachellel reviewed and looks good from here.

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,352 @@
+
+Relying Party application testing ensures that application modules operate as intended. In addition, application testing ensures that the system performs as expected and can properly process transactions that rely on FPKI certificates.
+
+This document does not define how to perform testing in CITE. That is a responsibility of the CITE Participatns. Some examples of testing conducted in CITE include:
```
```suggestion
This document does not define how to perform testing in CITE. That is a responsibility of the CITE Participants. Some examples of testing conducted in CITE include:
```

lachellel commented on _tools/cite-guide.md:
```diff
@@ -0,0 +1,352 @@
+This document does not define how to perform testing in CITE. That is a responsibility of the CITE Participatns. Some examples of testing conducted in CITE include:
+1. Interoperability testing between cross-certified Certification Authorities (CAs);
+2. Transition testing to new algorithms (e.g., SHA-2, ECC);
+3. PIV and PIV-I card interoperability testing;
```
```suggestion
3. PIV and PIV-I credential interoperability testing;
```

lachellel committed:
Update _tools/cite-guide.md

Co-Authored-By: weirdscience <kenneth.myers@protiviti.com>

lachellel committed:
Update _tools/cite-guide.md

Co-Authored-By: weirdscience <kenneth.myers@protiviti.com>

Commit:
Update test cert notice.

weirdscience commented:
@lachellel thank you for the updates. Looks good to merge.
